src/main/webapp/dev/js/main/feedbackSubmissionsEdit.es6

```diff
@@ -888,10 +888,10 @@ function updateRankMessageQn(qnNum) {
     }
 
     function checkMinMaxRestrictions(questionNumber, recipientIndex) {
-        const rankedOptions = $(`select[name="responsetext-${questionNumber}-${recipientIndex}"]`)
-                              .filter(function () {
-                                  return $(this).val() !== '';
-                              }).length;
+        const selector = $(`#rankToRecipients-${questionNumber}`).val() === 'true'
+                ? `select[name^="responsetext-${questionNumber}-"]`
```

Comment:
Looks like I missed this. What's the effect of selector with `^=` here?

Reply:
For a rank options question, each recipient has a set of dropdowns and the min/max restrictions impose on that set of dropdowns -

![capture](https://user-images.githubusercontent.com/11662001/31064122-2518f166-a757-11e7-8d1e-68d35f0311db.JPG)

But for rank recipients question, each recipient has just one drop-down, we impose our min/max restrictions over the set of all dropdowns for that question -

![capture](https://user-images.githubusercontent.com/11662001/31064242-1eeaf202-a758-11e7-84ed-623b828f526c.JPG)

so for selecting all dropdowns for the question we use `^=`. The different selectors are for selecting the required set of dropdowns.